### PR TITLE
Update CI, check everything runs up to py3.11

### PR DESCRIPTION
**Pull request by IAlibay**
Title says it all, also some QOL improvements on the build script.

### CODE REVIEW

**Comment by IAlibay** [comment]
Mostly double checking everything works upstream before we merge downstream.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,34 +9,38 @@ on:
...
+         mamba-version: "*"
+         add-pip-as-python-dependency: true
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
-        python -m pip install  pytest
-        conda install -c anaconda zlib
-        pip install -r requirements.txt
+        mamba install -c conda-forge zlib
+        mamba install --yes --file requirements.txt
```
Comment (IAlibay) [inline .github/workflows/build.yml:43]: @hmacdope do you mind if I just add a yaml file here to make life easier?

Comment (hmacdope) [reply .github/workflows/build.yml:43]: Not at all

**Comment by IAlibay** [comment]
Here we go, that should be good to go.

**Review by hmacdope — APPROVED** [review]
Thanks @IAlibay

**Pull request merged**